src/blocked.c

```diff
@@ -131,6 +132,13 @@ void processUnblockedClients(void) {
         listDelNode(server.unblocked_clients,ln);
         c->flags &= ~CLIENT_UNBLOCKED;
 
+        if (c->flags & CLIENT_MODULE) {
+            if (!(c->flags & CLIENT_BLOCKED)) {
+                moduleCallCommandUnblockedHandler(c);
```

Comment:
Question - can this be that a module called a command which was postponded? (in which case we will not execute it)

Reply:
Not following the question, do you mean that the unblock handler will call more blocking commands?

Reply:
i suppose postponed is in case it's a write command called during CLIENT PAUSE WRITE.
but these can't be MODULE type clients, right?

Reply:
That was basically my question. In case a module will call a command which will be postponed (either by having writes paused or by another busy module operation) Will in that case the module client will just terminate? or do we want it to be reprocessed?

Reply:
if the module's command is a write command it wouldn't have been executed, after it started, if an RM_Call gets blocked, it'll not get postponed (module fake client don't participate in that postpone game).
maybe that's an issue (module blocked clients violating write pause), but the reverse is impossible (a postponed (module) client that will then get blocked)

Reply:
@oranagra, I understand that postpone module fake clients will probably get paused by the original client executing the module command, but isn't it possible that module client be issued from a keyspace or other kind of event which is not write context (like key miss)? Maybe I am mixing and messing things up :) but I think that looking forward it is possible we would like to support more blocking cases which we would like to apply to module client, and I wanted to verify this will not hold us back. For example maybe we can introduce a processPendingCommand function that will process a command in case the pending flag is on and also call the the moduleCallCommandUnblockedHandler?

Reply:
I discussed this at length with Meir (he's adding a commit with a doc comment).
normally when an RM_Call gets blocked the module will block the original client as well, no writes will happen during failover pause, and the blocked client will remain blocked, and then the failover completed the demotion will abort the root client which the module should catch and abort the blocked fake client.
so other than documentation, i think the current code is good.

i'm not sure i understand your processPendingCommand suggestion above, maybe you can describe an example scenario you aim to handle, or suggest the exact code change you're proposing (maybe you did?)

Reply:
What I meant is that we can create the following function:
```
int processPendingCommand(client *c) {
    if (!(c->flags & CLIENT_PENDING_COMMAND || (c->flags & CLIENT_MODULE))
        return C_OK;
    client *old_client = server.current_client;
    server.current_client = c;
    enterExecutionUnit(1, 0);
    if (c->flags & CLIENT_PENDING_COMMAND) {
        c->flags &= ~CLIENT_PENDING_COMMAND;
        if (processCommandAndResetClient(c) == C_ERR) {
            return C_ERR;
        }
    }
    if (c->flags & CLIENT_MODULE) 
            moduleCallCommandUnblockedHandler(c);
    
    exitExecutionUnit();
    afterCommand(c);
    server.current_client = old_client;
```

So that we can use it in the context of unblockClientOnKey and processPendingCommandAndInputBuffer called from processUnblockedClients

Reply:
in any case according to what you explained, I do not recognize a bug with the current implementation, so we can also decide to refactor later on...